hip/matrix/ell_kernels.hip.cpp

```diff
@@ -111,40 +112,69 @@ using compiled_kernels = syn::value_list<int, 0, 1, 2, 4, 8, 16, 32>;
 namespace {
 
 
-template <int info, typename ValueType, typename IndexType>
+template <int dim, typename Type1, typename Type2>
+GKO_INLINE auto as_hip_accessor(
```

Comment:
Maybe this and the `as_cuda_accesor` can be moved to some common hip and common cuda files, as these will probably be useful for other mixed precision algorithms as well ?

Reply:
I will do this in PR #727:
https://github.com/ginkgo-project/ginkgo/blob/4408ac930431e9edf46cf1ebf5ef25e9e2dbd909/accessor/cuda_helper.hpp#L122